Source/JavaScriptCore/interpreter/Interpreter.cpp

```diff
@@ -451,14 +453,91 @@ bool Interpreter::isOpcode(Opcode opcode)
+
     AssertNoGC assertNoGC;
+
+    VM& vm = this->vm();
+    auto scope = DECLARE_THROW_SCOPE(vm);
+
+    auto getParentGenerator = [&](JSGenerator* gen) -> JSGenerator* {
+        JSValue contextValue = gen->internalField(static_cast<unsigned>(JSGenerator::Field::Context)).get();
+        JSPromise* awaitedPromise = jsDynamicCast<JSPromise*>(contextValue);
+        if (awaitedPromise && awaitedPromise->status(vm) == JSPromise::Status::Pending) {
+            JSValue reactionsValue = awaitedPromise->internalField(JSPromise::Field::ReactionsOrResult).get();
+            if (JSObject* reactions = jsDynamicCast<JSObject*>(reactionsValue)) {
+                // The @context field is always at offset 3 in Promise reactions objects
+                // created by pushNewPromiseReaction in PromiseOperations.js
+                constexpr PropertyOffset contextPropertyOffset = 3;
+                JSValue contextFieldValue = reactions->getDirect(contextPropertyOffset);
+                if (auto* resultGenerator = jsDynamicCast<JSGenerator*>(contextFieldValue))
+                    return resultGenerator;
+            }
+        }
+        return nullptr;
+    };
+
+    auto computeBytecodeIndex = [&](CodeBlock* codeBlock, JSGenerator* generator) -> BytecodeIndex {
+        BytecodeIndex bytecodeIndex(0);
+        JSValue stateValue = generator->internalField(static_cast<unsigned>(JSGenerator::Field::State)).get();
+        if (stateValue.isInt32()) {
+            int32_t state = stateValue.asInt32();
+            if (state > 0 && codeBlock->numberOfUnlinkedSwitchJumpTables() > 0) {
+                const UnlinkedSimpleJumpTable& jumpTable = codeBlock->unlinkedSwitchJumpTable(0);
+                int32_t offset = jumpTable.offsetForValue(state);
+                if (offset)
+                    bytecodeIndex = BytecodeIndex(offset);
+            }
+        }
+        return bytecodeIndex;
+    };
+
+    JSGenerator* currentGenerator = getParentGenerator(generator);
+    while (currentGenerator && results.size() < maxStackSize) {
+        JSValue nextValue = currentGenerator->internalField(static_cast<unsigned>(JSGenerator::Field::Next)).get();
+        JSFunction* asyncFunction = jsDynamicCast<JSFunction*>(nextValue);
+        if (asyncFunction && !asyncFunction->isHostOrBuiltinFunction()) {
+            if (FunctionExecutable* executable = asyncFunction->jsExecutable()) {
+                if (CodeBlock* codeBlock = executable->eitherCodeBlock()) {
+                    BytecodeIndex bytecodeIndex = computeBytecodeIndex(codeBlock, currentGenerator);
+                    results.append(StackFrame(vm, owner, asyncFunction, codeBlock, bytecodeIndex));
+                } else {
+                    // If CodeBlock doesn't exist, create a new one
+                    CodeBlock* newCodeBlock;
+                    executable->prepareForExecution<FunctionExecutable>(vm, asyncFunction, asyncFunction->scope(), CodeSpecializationKind::CodeForCall, newCodeBlock);
```

Comment:
We cannot allocate cells since these functions can be called from GC end phase.

Reply:
I changed it to display only the filename without showing line numbers if the codeBlock doesn't already exist.
https://github.com/WebKit/WebKit/pull/48314/files#diff-d32bbb01cc247b775900766b59d5d15e4db4ec21327c797d19d6dd66a7e7557dR504